astropy/io/fits/diff.py

```diff
@@ -149,21 +150,45 @@ def report(self, fileobj=None, indent=0, clobber=False):
             The number of 4 space tabs to indent the report.
 
         clobber : bool
-            Whether the report output should overwrite an existing file, when
-            fileobj is specified as a path.
+            If ``True`` and fileobj is a path, overwrite the output
+            file if it exist.
+            The use of clobber is deprecated; use overwrite instead.
+
+        overwrite : bool
+            If ``True`` and fileobj is a path, overwrite the output
+            file if it exist
+
+            If both clobber and overwrite are set to a boolean value
+            (whether equal or not), an exception is raised. If clobber
+            is used instead of overwrite, an AstropyDeprecationWarning
+            is issued.
 
         Returns
         -------
         report : str or None
+
         """
 
         return_string = False
         filepath = None
 
+        if isinstance(clobber, bool) and isinstance(overwrite, bool):
```

Comment:
Correct. It is a bit of a trade-off between the user doing the right thing (using booleans) and checking for every possibility (e.g., ``clobber=0`` equates to ``clobber=False``, not ``clobber=None``).
I felt that if someone uses ``clobber=1, overwrite=False``, it's up to them to live with the result. 
In cases where the value for clobber or overwrite derives from an evaluation, it'll generally be a bool, so such non-explicit values should give the correct result.